lib/active_node/validations.rb

```diff
@@ -34,6 +34,18 @@ module Validations
     autoload :UniquenessValidator, 'active_node/validations/uniqueness'
 
     module ClassMethods
+      # Initializes and saves an object in a single step.
+      def create(attributes = nil, &block)
+        if attributes.is_a?(Array)
+          attributes.collect { |attr| create(attr, &block) }
+        else
+          object = new(attributes)
+          yield(object) if block_given?
+          object.save
+          object
+        end
+      end
+
```

Comment:
Is this how they have it implemented in ActiveRecord? In an non DRY way, repeating nearly identical blocks?